packages/components/switch/src/Switch.vue

```diff
@@ -0,0 +1,95 @@
+<template>
+  <button
+    ref="switchNode"
+    class="ix-switch"
+    :class="[
+      { 'ix-switch-checked': stateChecked },
+      { 'ix-switch-disabled': disabled || loading },
+      { 'ix-switch-small': isSmallSize },
+    ]"
+    type="button"
+    @mouseup="handleMouseup"
+    @click="handleClick"
+  >
+    <div v-if="loading" class="ix-switch-loading">
+      <ix-icon name="loading" />
+    </div>
+    <div class="ix-switch-inner">
+      <template v-if="stateChecked">
+        <slot name="checkedChildren">{{ checkedChildren }}</slot>
+      </template>
+      <template v-else>
+        <slot name="unCheckedChildren">{{ unCheckedChildren }}</slot>
+      </template>
+    </div>
+  </button>
+</template>
+<script lang="ts">
+import { computed, defineComponent, ref, watch } from 'vue'
+import { SwitchProps } from './types'
+import { PropTypes } from '@idux/cdk/utils'
+import { IxIcon } from '@idux/components/icon'
+
+export default defineComponent({
+  name: 'IxSwitch',
+  components: { IxIcon },
+  inheritAttrs: false,
+  props: {
+    checked: PropTypes.bool.def(false),
+    disabled: PropTypes.bool.def(false),
+    checkedChildren: PropTypes.string.def(''),
+    unCheckedChildren: PropTypes.string.def(''),
+    size: PropTypes.oneOf(['normal', 'small'] as const).def('normal'),
+    loading: PropTypes.bool.def(false),
+  },
+  emits: ['change', 'update:checked'],
+  setup(props: SwitchProps, { emit }) {
+    const stateChecked = ref(props.checked ?? false)
+    const switchNode = ref<HTMLElement | null>(null)
+
+    const isSmallSize = computed(() => props.size === 'small')
+
+    watch(
+      () => props.checked,
+      curChecked => {
+        stateChecked.value = curChecked ?? false
+      },
+    )
+
+    const setChecked = (checked: boolean, e: Event) => {
+      stateChecked.value = !checked
+      emit('update:checked', stateChecked.value, e)
```

Comment:
```suggestion
      emit('update:checked', stateChecked.value)
```
双向绑定的 emit 只能有一个参数，你可以注意下控制台的警告